CMakeLists.txt

```diff
@@ -185,8 +185,12 @@ if(OPENMP_ENABLE)
   #message(STATUS "****** finding OpenMP support")
   FIND_PACKAGE(OpenMP)
   if(OPENMP_FOUND)
-    add_compile_options($<$<COMPILE_LANGUAGE:C>:${OpenMP_C_FLAGS}> $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>)
-    link_libraries(-fopenmp)
+    if(${CMAKE_VERSION} VERSION_GREATER "3.8.9")
+      link_libraries(OpenMP::OpenMP_C OpenMP::OpenMP_CXX)
+    else()
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
```

Comment:
I think this part should be using add_compile_options and link_libraries since we want to avoid (ab)using the CMAKE_XXX_FLAGS variables  and I think we do need the link option otherwise this isn't going to work on older linux systems with older cmake.

Reply:
So maybe something like:
```
add_compile_options($<$<COMPILE_LANGUAGE:C>:${OpenMP_C_FLAGS}> $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>)
link_libraries($<$<COMPILE_LANGUAGE:C>:${OpenMP_C_FLAGS}> $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>)
```

Reply:
Pretty much all of the builds are using cmake 3.4 or 3.5 on Travis (whichever the minium required is).